qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/commons/JavaElementInfo.java

```diff
@@ -112,33 +110,27 @@ public String getJavaElementSimpleType() {
 	 * @return the simple type of the given type.
 	 */
 	public static String getSimpleType(String type) {
-		if (type == null) {
-			return null;
+		if (type == null || type.indexOf('.') == -1) {
+			// ex: List
+			return type;
 		}
-		int startBracketIndex = type.indexOf('<');
-		if (startBracketIndex != -1) {
-			int endBracketIndex = type.indexOf('>', startBracketIndex);
-			// Main type
-			StringBuilder simpleType = new StringBuilder(
-					getSimpleTypeWithoutGeneric(type.substring(0, startBracketIndex)));
-			// Generic type
-			simpleType.append('<');
-			String[] generics = type.substring(startBracketIndex + 1, endBracketIndex).split(",");
-			StringJoiner commaJoiner = new StringJoiner(",");
-			for (String generic : generics) {
-				commaJoiner.add(getSimpleTypeWithoutGeneric(generic));
+		StringBuilder result = new StringBuilder();
+		boolean parsingClassName = false;
+		for (int i = 0; i < type.length(); i++) {
+			char c = type.charAt(i);
+			if (!parsingClassName && Character.isUpperCase(c)) {

```

Comment:
Similar comment about the class name

Reply:
See https://github.com/redhat-developer/quarkus-ls/pull/743#discussion_r980219970

Reply:
fixed, now I'm using $